src/Tools/dotnet-monitor/DiagnosticPortValidateOptions.cs

```diff
@@ -12,8 +14,28 @@ namespace Microsoft.Diagnostics.Tools.Monitor
     internal class DiagnosticPortValidateOptions :
         IValidateOptions<DiagnosticPortOptions>
     {
+        private readonly CollectionRulesConfigurationProvider _provider;
+        private readonly ILogger<DiagnosticPortValidateOptions> _logger;
+
+        public DiagnosticPortValidateOptions(CollectionRulesConfigurationProvider provider, ILogger<DiagnosticPortValidateOptions> logger)
+        {
+            _provider = provider;
+            _logger = logger;
+        }
+
         public ValidateOptionsResult Validate(string name, DiagnosticPortOptions options)
         {
+            var collectionRuleNames = _provider.GetCollectionRuleNames();
+
+            if (collectionRuleNames != null && collectionRuleNames.Count > 0)
+            {
+                if (options.ConnectionMode != DiagnosticPortConnectionMode.Listen
+                    || string.IsNullOrEmpty(options.EndpointName))
+                {
+                    _logger.LogWarning(Strings.ErrorMessage_DiagnosticPortNotInListenModeForCollectionRules);
```

Comment:
I recognize that this isn't the same as the actual validation that follows it, and so it maybe shouldn't be in this file - in the meantime I left it here due to its similar nature, but something different can be done if this feels incorrect.